packages/neuron-wallet/src/services/wallets.ts

```diff
@@ -34,33 +34,9 @@ export default class WalletService {
     return key.checkPassword(password)
   }
 
-  // TODO: update wallet
-  // public update = ({
-  //   id,
-  //   name,
-  //   address,
-  //   publicKey,
-  // }: {
-  //   id: string
-  //   name?: string
-  //   address?: string
-  //   publicKey?: Uint8Array
-  // }): boolean => {
-  //   const wallet = this.show(id)
-  //   if (wallet) {
-  //     if (name) {
-  //       wallet.name = name
-  //     }
-  //     if (address) {
-  //       wallet.address = address
-  //     }
-  //     if (publicKey) {
-  //       wallet.publicKey = publicKey
-  //     }
-  //     return true
-  //   }
-  //   return false
-  // }
+  public update = (walletId: string, newWallet: WalletData) => {
+    walletStore.update(walletId, newWallet)
```

Comment:
It seems that `undefined` not handled in this case.
if `newWallet.name` is undefined, it's current name should be used.

```javascript
{ ...oldWallet, ...newWallet }
```
will cover older wallet except the undefined fields.

Reply:
get it